ibm/resource_ibm_database.go

```diff
@@ -0,0 +1,1086 @@
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev1/controller"
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev1/management"
+	"github.com/IBM-Cloud/bluemix-go/models"
+
+	"github.com/IBM-Cloud/bluemix-go/bmxerror"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	validation "github.com/hashicorp/terraform/helper/validation"
+)
+
+const (
+	databaseInstanceSuccessStatus  = "active"
+	databaseInstanceProgressStatus = "in progress"
+	databaseInstanceInactiveStatus = "inactive"
+	databaseInstanceFailStatus     = "failed"
+	databaseInstanceRemovedStatus  = "removed"
+)
+
+const (
+	databaseTaskSuccessStatus  = "completed"
+	databaseTaskProgressStatus = "running"
+	databaseTaskFailStatus     = "failed"
+)
+
+type CsEntry struct {
+	Name       string
+	String     string
+	CertName   string
+	CertBase64 string
+}
+
+func resourceIBMDatabaseInstance() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMDatabaseInstanceCreate,
+		Read:     resourceIBMDatabaseInstanceRead,
+		Update:   resourceIBMDatabaseInstanceUpdate,
+		Delete:   resourceIBMDatabaseInstanceDelete,
+		Exists:   resourceIBMDatabaseInstanceExists,
+		Importer: &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(40 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Resource instance name for example, my Database instance",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+
+			"resource_group_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The id of the resource group in which the Database instance is present",
+			},
+
+			"location": {
+				Description: "The location or the environment in which Database instance exists",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+
+			"service": {
+				Description:  "The name of the Cloud Internet database service",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAllowedStringValue([]string{"databases-for-etcd", "databases-for-postgresql", "databases-for-redis", "databases-for-elasticsearch", "databases-for-mongodb", "messages-for-rabbitmq", "databases-for-mysql"}),
+			},
+			"plan": {
+				Description:  "The plan type of the Database instance",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAllowedStringValue([]string{"standard"}),
+			},
+
+			"status": {
+				Description: "The resource instance status",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"adminuser": {
+				Description: "The admin user id for the instance",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"adminpassword": {
+				Description:  "The admin user id for the instance",
```

Comment:
Description shd be "The admin user passwd..."...If user doesn't provide password will the API set back any password?

Reply:
Corrected text. If not set by the user, I believe the Id is locked. There is no default password. I've updated the docs to say that another user must be added.